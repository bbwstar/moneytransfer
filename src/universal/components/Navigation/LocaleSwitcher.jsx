import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import Cookie from 'js-cookie';

class LocaleSwitcher extends Component {
  handleClick = () => {
    Cookie.set('locale', this.props.intl.locale === 'en' ? 'cs' : 'en');
    window.location.reload();
  };

  render() {
    return (
      <a className="nav-item" onClick={this.handleClick} role="menuitem" tabIndex="0">
        {this.props.intl.locale === 'en' ? 'Čeština' : 'English'}
      </a>
    );
  }
}

LocaleSwitcher.propTypes = {
  intl: PropTypes.shape({
    locale: PropTypes.string.isRequired,
  }).isRequired,
};

export default injectIntl(LocaleSwitcher);
