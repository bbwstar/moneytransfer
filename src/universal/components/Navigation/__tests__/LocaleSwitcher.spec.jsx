import React from 'react';
import Cookie from 'js-cookie';

import createComponentWithIntl from 'utils/intl-renderer-test-helper';
import LocaleSwitcher from '../LocaleSwitcher';

describe('LocaleSwitcher (Snapshot)', () => {
  it('renders and changes locale from en to cs', () => {
    const component = createComponentWithIntl(
      <LocaleSwitcher />,
      { locale: 'en' },
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();

    window.location.reload = jest.fn();
    Cookie.set = jest.fn();
    tree.props.onClick();

    expect(window.location.reload).toBeCalled();
    expect(Cookie.set).toBeCalledWith('locale', 'cs');
  });

  it('renders and changes locale from cs to en', () => {
    const component = createComponentWithIntl(
      <LocaleSwitcher />,
      { locale: 'cs' },
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();

    window.location.reload = jest.fn();
    Cookie.set = jest.fn();
    tree.props.onClick();

    expect(window.location.reload).toBeCalled();
    expect(Cookie.set).toBeCalledWith('locale', 'en');
  });
});

