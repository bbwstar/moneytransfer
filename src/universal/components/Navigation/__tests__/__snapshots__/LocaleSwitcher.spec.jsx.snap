// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleSwitcher (Snapshot) renders and changes locale from cs to en 1`] = `
<a
  className="nav-item"
  onClick={[Function]}
  role="menuitem"
  tabIndex="0"
>
  English
</a>
`;

exports[`LocaleSwitcher (Snapshot) renders and changes locale from en to cs 1`] = `
<a
  className="nav-item"
  onClick={[Function]}
  role="menuitem"
  tabIndex="0"
>
  Čeština
</a>
`;
